doc/PerformanceTesting.md

```diff
@@ -0,0 +1,169 @@
+
+### Location
+
+Performance tests are located under **"test/perf"** folder under the service folder(e.g. alongside ut folder).
+
+The perf test folder should be added in the parent CMakeLists.txt guarded by the **BUILD_PERF_TEST** flag.
+
+E.G.
+
+```markdown
+
+    if(BUILD_PERFORMANCE_TESTS)
+        add_subdirectory(test/perf)
+    endif()
+
+```
+
+### Structure
+
+As any other CPP project you will need a **CMakeLists.txt** file , along side **src** and **inc** folders
+
+#### Inc
+
+Under inc folder create a subfolder structure following your namespace hierarchy ending with another test folder since these are in the ...Test namespace.
+
+E.G. test/perf/inc/azure/storage/blobs/test
+
+The tests are defined in .hpp files under this folder. 
+
+##### Test definition
+
+In a .hpp file we define a class that will be contain the tests methods. The class is defined in the ::Test namespace for your service. 
+
+The class will inherit from the **PerfTest** base  class and will override several methods as follows:
+- Constructor(Azure::Perf::TestOptions options) : PerfTest(options)
+  Options field are passed from the perf test framework and constain the various options defined for running the test.
+- void Run(Azure::Core::Context const&) override {...}.
+  Runs the actual test code. We recommend doing as little extra work here besides the actual relevant code to be measured. Avoid assert, if etc methods as ant extra work will skew the data.
+- std::vector<Azure::Perf::TestOption> GetTestOptions() override 
+  Defines the various parameters for the test run that can be passed to the test from the performance framework. The perf framework uses these params to run various combinations(e.g. blob size)
+- static Azure::Perf::TestMetadata GetTestMetadata()
+  Returns TestMetadate object used to identify the test. 
+
+#### Src 
+
+Contains one cpp file that contains the main method defintion 
+
+```cpp
+    int main(int argc, char** argv)
+    {
+    std::cout << "SERVICE VERSION " << VCPKG_SERVICE_VERSION << std::endl;
+    // Create the test list
+    std::vector<Azure::Perf::TestMetadata> tests{
+        Service::Namespace::Test::TestName::GetTestMetadata(),
+    };
+    
+    Azure::Perf::Program::Run(Azure::Core::Context::ApplicationContext, tests, argc, argv);
+
+    return 0;
+    }
+```
+
+#### CMakeLists.txt
+
+Beyond the regular cmake defintion in your cmake file make sure to add 
+
+```makefile
+include(AzureVcpkg)
+az_vcpkg_integrate()
+...
+include(PerfTest)
+SETPERFDEPS(azure-storage-blobs-cpp VCPKG_SERVICE_VERSION)
+```
+
+The crucial part here is the SetPerfDeps cmake macro. \
+The perf framework will set an environment variable based on the service name with the value representing a version, thus allowing to run the tests against diffrent VCPKG published versions. If the env is not defined then the test will build against the current source code of the service. 
```

Comment:
What is the environment variable name?

Reply:
it does not concern the user, the thing is set up so theat you don't have to know or care what the env is

Reply:
If it doesn't matter to the customer, why mention it?